### PR TITLE
HTBHF-632 Added input validation.

### PR DESCRIPTION
**Pull request by dwybourn**

### CODE REVIEW

**Comment by dwybourn** [comment]
Initial PR to decided on approach for validation tests. We can use spock to create a large set of parameterised validation tests, create a method for each test (as done in this PR), or use junit 5 parameterisation. 

Note that junit5 parameterised tests do not yet support null values and the setting of fields as done in the claimant service would look a lot messier in java than groovy.

**Comment by YetAnotherMatt** [comment]
@dwybourn maybe a combination of JUNIT5 parameterised tests, plus individual tests for null values?

File: api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java [hunk 1]
```diff
@@ -42,4 +52,60 @@ void shouldGetEligibility() {
...
         assertThat(response.getBody()).isEqualTo(eligibilityResponse);
     }
+
+    @Test
+    void shouldReturnBadRequestForMissingNino() {
+        PersonDTO person = aPersonWithNoNino();
+
+        var benefit = restTemplate.postForEntity(ENDPOINT_URL, person, EligibilityResponse.class);
+
+        Assertions.assertThat(benefit.getStatusCode()).isEqualTo(BAD_REQUEST);
```
Comment (mikeSmith08) [inline api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java]: You don't need the `Assertions.` on all your `assertThat` method calls in this test class.

Comment (dwybourn) [reply api/src/test/java/uk/gov/dhsc/htbhf/eligibility/controller/EligibilityControllerTest.java]: Removed.

**Pull request merged**